v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/GcpToNeo4j.java

```diff
@@ -0,0 +1,393 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.neo4j;
+
+import com.google.cloud.teleport.v2.neo4j.actions.ActionBeamFactory;
+import com.google.cloud.teleport.v2.neo4j.actions.ActionFactory;
+import com.google.cloud.teleport.v2.neo4j.actions.preload.PreloadAction;
+import com.google.cloud.teleport.v2.neo4j.database.Neo4jConnection;
+import com.google.cloud.teleport.v2.neo4j.model.InputRefactoring;
+import com.google.cloud.teleport.v2.neo4j.model.InputValidator;
+import com.google.cloud.teleport.v2.neo4j.model.connection.ConnectionParams;
+import com.google.cloud.teleport.v2.neo4j.model.enums.ActionExecuteAfter;
+import com.google.cloud.teleport.v2.neo4j.model.enums.ArtifactType;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.JobSpecMapper;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.OptionsParamsMapper;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.SourceQuerySpec;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.TargetQuerySpec;
+import com.google.cloud.teleport.v2.neo4j.model.job.Action;
+import com.google.cloud.teleport.v2.neo4j.model.job.ActionContext;
+import com.google.cloud.teleport.v2.neo4j.model.job.JobSpec;
+import com.google.cloud.teleport.v2.neo4j.model.job.OptionsParams;
+import com.google.cloud.teleport.v2.neo4j.model.job.Source;
+import com.google.cloud.teleport.v2.neo4j.model.job.Target;
+import com.google.cloud.teleport.v2.neo4j.providers.Provider;
+import com.google.cloud.teleport.v2.neo4j.providers.ProviderFactory;
+import com.google.cloud.teleport.v2.neo4j.transforms.GcsLogTransform;
+import com.google.cloud.teleport.v2.neo4j.transforms.Neo4jRowWriterTransform;
+import com.google.cloud.teleport.v2.neo4j.utils.BeamBlock;
+import com.google.cloud.teleport.v2.neo4j.utils.ModelUtils;
+import com.google.cloud.teleport.v2.neo4j.utils.ProcessingCoder;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.Wait;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Dataflow template which reads BigQuery data and writes it to Neo4j. The source data can be either
+ * a BigQuery table or an SQL query.
+ */
+public class GcpToNeo4j {
```

Comment:
We use `googlecloud` to have it grouped, but probably this pipeline itself could be called `BigQueryToNeo4j`